### PR TITLE
[libc] Temporary math macros fix

### PR DESCRIPTION
**Pull request by michaelrj-google**
Downstream's having some issues due to math-macros.h issues. These will
be fixed properly soon.

See https://github.com/llvm/llvm-project/issues/87683 for tracking this tech debt.

### CODE REVIEW

**Review by nickdesaulniers — APPROVED** [review]
Please link to https://github.com/llvm/llvm-project/issues/87683 in the comment and commit message. Please wait for @lntue 's feedback before landing.

File: libc/include/llvm-libc-macros/math-macros.h
Comment (nickdesaulniers) [reply libc/include/llvm-libc-macros/math-macros.h:13]: oh, right, we don't define `LIBC_FULL_BUILD` yet until (https://github.com/llvm/llvm-project/pull/87598, https://github.com/llvm/llvm-project/pull/87017)

**Comment by michaelrj-google** [comment]
I will land this after the presubmits finish.

**Review by lntue — APPROVED** [review]

**Review by nickdesaulniers — APPROVED** [review]

**Pull request merged**
